### PR TITLE
Use ICEBERG_MISSING_METADATA error code in IcebergSplitSource

### PR DESCRIPTION
**Pull request by losipiuk**
## Description

    Previously if we got exception pointing out to missing metadata files
    during split enumeration query was failing with GENERIC_INTERNAL_ERROR
    instead of more appropriate, Iceberg specific error code.

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergExceptions.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import java.io.FileNotFoundException;
+
+public final class IcebergExceptions
+{
+    private IcebergExceptions() {}
+
+    public static boolean isNotFoundException(Throwable failure)
+    {
+        // qualified name, as this is NOT the io.trino.spi.connector.NotFoundException
+        return failure instanceof org.apache.iceberg.exceptions.NotFoundException ||
```
Comment (wendigo) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergExceptions.java:25]: getCasualChain().stream().anyMatch ?

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/IcebergQueryRunner.java [hunk 2]
```diff
@@ -479,13 +479,20 @@ private IcebergQueryRunnerWithTaskRetries() {}
         public static void main(String[] args)
                 throws Exception
         {
-            Path exchangeManagerDirectory = createTempDirectory(null);
+            Logger log = Logger.get(IcebergQueryRunnerWithTaskRetries.class);
```
Comment (raunaqmorarka) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/IcebergQueryRunner.java:482]: Does it matter whether the logger is created here or as it was before ?

**Review by raunaqmorarka — APPROVED** [review]

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/IcebergQueryRunner.java (hunk 2 above)
Comment (losipiuk) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/IcebergQueryRunner.java:482]: It does not - just consistent with other classes.

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergExceptions.java (hunk 1 above)
Comment (losipiuk) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergExceptions.java:25]: makes sense

Comment (losipiuk) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergExceptions.java]: I cleaned it up a bit more. PTAL @raunaqmorarka and @wendigo

**Comment by findinpath** [comment]
Can you pls add a stacktrace to the description of the PR to understand better why these changes are made?

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergExceptions.java [hunk 3]
```diff
@@ -13,18 +13,44 @@
  */
```
Comment (raunaqmorarka) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergExceptions.java:13]: typo in commit message `hadling`

**Review by raunaqmorarka — APPROVED** [review]

**Comment by losipiuk** [comment]
> Can you pls add a stacktrace to the description of the PR to understand better why these changes are made?

Added some more description

**Comment by losipiuk** [comment]
> No release notes entry @losipiuk ??

Nope

**Pull request merged**